src/Components/Web/src/Forms/InputLargeTextArea/InputLargeTextArea.cs

```diff
@@ -0,0 +1,149 @@
+        /// </summary>
+        [Parameter(CaptureUnmatchedValues = true)]
+        public IDictionary<string, object>? AdditionalAttributes { get; set; }
+
+        /// <summary>
+        /// Gets or sets the associated <see cref="ElementReference"/>.
+        /// <para>
+        /// May be <see langword="null"/> if accessed before the component is rendered.
+        /// </para>
+        /// </summary>
+        [DisallowNull]
+        public ElementReference? Element
+        {
+            get => _inputLargeTextAreaElement;
+            protected set => _inputLargeTextAreaElement = value!.Value;
+        }
+
+        /// <inheritdoc/>
+        protected override async Task OnAfterRenderAsync(bool firstRender)
+        {
+            if (firstRender)
+            {
+                _jsCallbacksRelay = new InputLargeTextAreaJsCallbacksRelay(this);
+                await JSRuntime.InvokeVoidAsync(InputLargeTextAreaInterop.Init, _jsCallbacksRelay.DotNetReference, _inputLargeTextAreaElement);
+            }
+        }
+
+        /// <inheritdoc/>
+        protected override void BuildRenderTree(RenderTreeBuilder builder)
+        {
+            builder.OpenElement(0, "textarea");
+            builder.AddMultipleAttributes(1, AdditionalAttributes);
+            builder.AddElementReferenceCapture(2, elementReference => _inputLargeTextAreaElement = elementReference);
+            builder.CloseElement();
+        }
+
+        Task IInputLargeTextAreaJsCallbacks.NotifyChange(int length)
+            => OnChange.InvokeAsync(new InputLargeTextAreaChangeEventArgs(this, length));
+
+        /// <summary>
+        /// Retrieves the textarea value asyncronously.
+        /// </summary>
+        /// <param name="maxLength">The maximum length of content to fetch from the textarea.</param>
+        /// <param name="cancellationToken">The <see cref="System.Threading.CancellationToken"/> used to relay cancellation of the request.</param>
+        /// <returns>A <see cref="System.IO.TextReader"/> which facilitates reading of the textarea value.</returns>
+        public async ValueTask<StreamReader> GetTextAsync(int maxLength = 32_000, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                var streamRef = await JSRuntime.InvokeAsync<IJSStreamReference>(InputLargeTextAreaInterop.GetText, cancellationToken, _inputLargeTextAreaElement);
+                var stream = await streamRef.OpenReadStreamAsync(maxLength, cancellationToken);
+                var streamReader = new StreamReader(stream);
+                return streamReader;
+            }
+            catch (JSException jsException)
+            {
+                // Special casing support for empty textareas. Due to security considerations
+                // 0 length streams/textareas aren't permitted from JS->.NET Streaming Interop.
+                if (jsException.InnerException is ArgumentOutOfRangeException)
```

Comment:
Can we throw a more specific exception? Hard to say that `ArgumentOutOfRange` is specifically because of a 0-length stream?

Reply:
Hmm in the original context, I believe the `ArgumentOutOfRange` exception may make sense, however I see your point about it still being ambiguous with other areas where that exception may be thrown.

How about this:
https://github.com/dotnet/aspnetcore/pull/34856/files#diff-30d94608ca41ec93dcbe8eb2bb646257519e86080e81ed5c246a715a8782c4b9R94-R102

Alternatively, we can decorate the `Exception.Data` property with something more specific (though that may not necessarily be statically typed).